src/main/java/vehicle/Vehicle.java

```diff
@@ -106,6 +106,45 @@ public static VehicleBuilder builder(String brand, String model, double price, i
         return new VehicleBuilder(brand, model, price, id);
     }
 
+    /**
+     * Calcola il prezzo scontato e modifica la variabile discountedPrice.
+     * Inserisce un double che viene convertito internamente in un BigDecimal.
+     *
+     * @param discountPercentage è la percentuale di sconto che si desidera applicare
+     * @throws ExcessiveParameterException se la percentuale inserita è fuori dai limiti 0 e 100
+     */
+
+    protected void calculateDiscount(double discountPercentage) throws ExcessiveParameterException {
+        if (discountPercentage > 100 || discountPercentage < 0) {
+            throw new ExcessiveParameterException("The discount percentage must be comprehended between 0 and 100");
+        }
+        BigDecimal discountRate = BigDecimal.valueOf(discountPercentage / 100).setScale(2, RoundingMode.HALF_EVEN);
+        BigDecimal removedPrice = originalPrice.multiply(discountRate).setScale(2, RoundingMode.HALF_EVEN);
+        discountedPrice = originalPrice.subtract(removedPrice).setScale(2, RoundingMode.HALF_EVEN);
+    }
+
+    /**
+     * Permette di decidere se attivare uno sconto e di scegliere di quanto scontare il prodotto.
+     *
+     * @param discountPercentage è la percentuale di sconto che si desidera applicare.
+     */
+    public void activateDiscount(double discountPercentage) {
+        discountFlag = true;
+        try {
+            calculateDiscount(discountPercentage);
+        } catch (ExcessiveParameterException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
```

Comment:
credo che in activateDiscount non assegni il prezzo a discountedPrice

Reply:
È un compito di "calculate discount" il metodo richiamato in activate discount. Nei test puoi vedere che modifica proprio quella variabile

Reply:
tutto chiaro